js/battle.js

```diff
@@ -6711,6 +6711,9 @@ var Battle = (function () {
 			poke.removeVolatile('typechange');
 
 			var newSpecies = args[2];
+			if (!newSpecies) {
+				poke.sprite.animTransform($.extend(spriteData, Tools.getTemplate(poke)), true);
+			}
```

Comment:
If all the rest is wrong, this is actually a new protocol added in the server PR in case of an Pokemon with an active Illusion undergoing a permanent forme change so that it does a `'-detailschange'` without actually changing details. The current protocol supports a details change that doesn't actually change anything but still does an animation that doesn't necessarily reveal it as an Illusion, but this allows the server to effectively tell the client ahead of time that there's nothing to actually change and to just play the animation.